src/Compilers/Core/MSBuildTask/LocateEditorConfigFiles.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public sealed class LocateEditorConfigFiles : Task
+    {
+        [Required]
+        public ITaskItem[] InputFiles { get; set; }
+
+        [Output]
+        public ITaskItem[] EditorConfigFiles { get; private set; }
+
+        public LocateEditorConfigFiles()
+        {
+            TaskResources = ErrorString.ResourceManager;
+        }
+
+        public override bool Execute()
+        {
+            try
+            {
+                ExecuteCore();
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Log.LogErrorFromException(ex);
+                return false;
+            }
+        }
+
+        private void ExecuteCore()
+        {
+            var directoriesAddedToQueue = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
@jaredpar I chose this because this was the same comparer used for the CS2002 "source file is specified multiple times" warning, which was the only other place I could think the compiler would be explicitly comparing paths. Is that a bug there too, or should I add a comment that for some reason those are different?